pkg/server/backuprestoreserver.go

```diff
@@ -185,6 +188,10 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 			}
 			go b.runEtcdProbeLoopWithSnapshotter(leCtx, handler, ssr, ss, ssrStopCh, ackCh)
 			go defragmentor.DefragDataPeriodically(leCtx, b.config.EtcdConnectionConfig, b.defragmentationSchedule, defragCallBack, b.logger)
+			//start etcd member garbage collector
+			if b.config.HealthConfig.MemberGCEnabled {
+				go membergarbagecollector.RunMemberGarbageCollectorPeriodically(ctx, memGCStopCh, b.config.HealthConfig, b.logger, b.config.EtcdConnectionConfig)
```

Comment:
Can't we pass `leCtx` instead of using both `ctx` and `memGCStopCh`? `leCtx` is cancelled when leader election is lost (see [here](https://github.com/gardener/etcd-backup-restore/blob/322d3c74fcd5ea7be9961ce5e9d7ffed48ecc34e/pkg/leaderelection/leaderelection.go#L135)).

Reply:
Thanks for pointing this out. Updated to use only `leCtx` now